### PR TITLE
chore: Remove need for dockerfile

### PR DESCRIPTION
**Pull request by TylerHelmuth**
## Which problem is this PR solving?

Currently whenever we want to bump the timestampprocessor's libraries or build a new version of our distribution of the collector with the latest components we have to build and push a dockerfile to DockerHub. 

## Short description of the changes
To avoid having to push to dockerhub, I have updated the CircleCI workflow to install the dependencies itself.

## How to verify that this has the expected result
Workflow passed

### CODE REVIEW

**Review by kentquirk — APPROVED** [review]
I like the idea here, but if we can make it less brittle by leaving off patches from version numbers, we should.

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,15 +3,39 @@ jobs:
...
+      - run:
+          name: install dependencies
+          command: |
+            sudo apt-get update -qq
+            sudo apt-get install -y python2  libbz2-dev  libreadline-dev libssl-dev openssl
+            curl https://pyenv.run | bash 
+            sudo rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+            export PYENV_ROOT=/home/circleci/.pyenv
+            export PATH=/home/circleci/.pyenv/shims:/home/circleci/.pyenv/bin:/home/circleci/.poetry/bin:$PATH
+            pyenv install 3.9.12
```
Comment (kentquirk) [inline .circleci/config.yml:21]: Can we make these two use a single envvar, preferably without the patch specified?

Comment (TylerHelmuth) [reply .circleci/config.yml]: Done

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,15 +3,39 @@ jobs:
   build:
     working_directory: ~/repo
     docker:
-      - image: honeycombio/cci-go-yq:0.0.8
+      - image: cimg/go:1.19.4
```
[the remark this answers is not in the document]
Comment (TylerHelmuth) [reply .circleci/config.yml]: done

**Pull request merged**